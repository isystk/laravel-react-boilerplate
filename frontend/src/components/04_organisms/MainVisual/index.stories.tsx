import { Meta, Story } from '@storybook/react'
import React from 'react'
import MainVisual from './index'

export default {
  title: '04_organisms/MainVisual',
  component: MainVisual,
} as Meta

const Template: Story = () => {
  return <MainVisual />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
