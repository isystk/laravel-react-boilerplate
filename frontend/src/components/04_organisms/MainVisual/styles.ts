import { css, keyframes } from '@emotion/css'

export const mainVisual = css``
