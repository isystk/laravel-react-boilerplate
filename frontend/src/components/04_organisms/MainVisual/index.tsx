import React, { FC } from 'react'
import { ContainerProps } from 'types'
import * as styles from './styles'
import { connect } from '@/components/hoc'
import LineButton from '@/components/01_atoms/LineButton'
import Image from '@/components/01_atoms/Image'
import { useI18n } from '@/components/i18n'
import ScrollIn from '@/components/02_interactions/ScrollIn'

/** MainVisualProps Props */
export type MainVisualProps = Record<never, never>
/** Presenter Props */
export type PresenterProps = MainVisualProps & {
  t
}

/** Presenter Component */
const MainVisualPresenter: FC<PresenterProps> = ({ t, ...props }) => (
  <>
    <section
      className={`${styles.mainVisual}`}
      style={{ height: 'calc(100vh - 4rem)' }}
    >
      <div className="flex flex-wrap items-center h-full bg-main justify-center">
        <div className="flex w-full md:w-1/2 justify-center md:justify-end md:pr-16">
          <Image
            src="/images/iphone.png"
            className="w-36 md:w-72"
            alt="Line App"
            lazy={false}
          />
        </div>
        <div className="flex w-full md:w-1/2 justify-center md:justify-start pb-8">
          <div className="px-8">
            <ScrollIn delay="0s">
              <p className="text-accent text-xl md:text-3xl font-bold mb-6 ">
                {t('One million registered within one month of release!')}
              </p>
            </ScrollIn>
            <ScrollIn delay="1s">
              <h1 className="text-gray-700 text-2xl md:text-4xl font-bold mb-6 ">
                {t('The topic of ________. Available on line!')}
              </h1>
            </ScrollIn>
            <ScrollIn delay="2s">
              <LineButton
                link="#"
                label={t('Add me as a friend and ask me a question')}
              />
            </ScrollIn>
          </div>
        </div>
      </div>
    </section>
  </>
)

/** Container Component */
const MainVisualContainer: React.FC<
  ContainerProps<MainVisualProps, PresenterProps>
> = ({ presenter, ...props }) => {
  const { t } = useI18n('Common')
  return presenter({
    t,
    ...props,
  })
}

export default connect<MainVisualProps, PresenterProps>(
  'MainVisual',
  MainVisualPresenter,
  MainVisualContainer
)
