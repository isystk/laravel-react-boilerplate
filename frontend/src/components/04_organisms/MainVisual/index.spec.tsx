import React from 'react'
import renderer from 'react-test-renderer'
import MainVisual from './index'

describe('MainVisual', () => {
  it('Match Snapshot', () => {
    const component = renderer.create(<MainVisual />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
