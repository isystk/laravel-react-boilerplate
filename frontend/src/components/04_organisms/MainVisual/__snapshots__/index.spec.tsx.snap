// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainVisual Match Snapshot 1`] = `
<section
  className="css-0"
  style={
    {
      "height": "calc(100vh - 4rem)",
    }
  }
>
  <div
    className="flex flex-wrap items-center h-full bg-main justify-center"
  >
    <div
      className="flex w-full md:w-1/2 justify-center md:justify-end md:pr-16"
    >
      <div
        className="w-36 md:w-72"
      >
        <img
          alt="Line App"
          className="css-s08p0c "
          src="/images/iphone.png"
        />
      </div>
    </div>
    <div
      className="flex w-full md:w-1/2 justify-center md:justify-start pb-8"
    >
      <div
        className="px-8"
      >
        <div
          className="undefined css-pjt9eq css-1exhw58"
        >
          <p
            className="text-accent text-xl md:text-3xl font-bold mb-6 "
          >
            One million registered within one month of release!
          </p>
        </div>
        <div
          className="undefined css-pjt9eq css-xrumgy"
        >
          <h1
            className="text-gray-700 text-2xl md:text-4xl font-bold mb-6 "
          >
            The topic of ________. 
            <br />
             Available on line!
          </h1>
        </div>
        <div
          className="undefined css-pjt9eq css-6y95dw"
        >
          <a
            className="css-0 p-2 pl-4 pr-4 flex items-center justify-center bg-accent rounded-full"
            href="#"
          >
            <img
              alt="Add me as a friend and ask me a question"
              height="50px"
              src="/images/icon-line.png"
              width="50px"
            />
            <span
              className="text-white font-bold"
            >
              Add me as a friend and ask me a question
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
